### PR TITLE
ISSUE-1253 Transaction composable implementation

### PR DESCRIPTION
**Pull request by Milan-Nerevar**
Snímka obrazovky 2024-06-04 o 16 33 40

### CODE REVIEW

File: demo-app/src/main/java/cash/z/ecc/android/sdk/demoapp/ui/screen/transactions/view/TransactionsView.kt [hunk 1]
```diff
@@ -23,63 +21,91 @@ import androidx.compose.runtime.Composable
...
-    MaterialTheme {
-        // TODO [#1090]: Demo: Add Addresses and Transactions Compose Previews
-        // TODO [#1090]: https://github.com/zcash/zcash-android-wallet-sdk/issues/1090
-        // TransactionsView()
+fun Transactions(synchronizer: Synchronizer, onBack: () -> Unit) {
+    val queryScope = rememberCoroutineScope()
+
+    // TODO [#846]: Slow addresses providing
+    // TODO [#846]: https://github.com/zcash/zcash-android-wallet-sdk/issues/846
+    val walletAddresses: WalletAddresses? = flow<WalletAddresses?> {
```
Comment (Milan-Nerevar) [inline demo-app/src/main/java/cash/z/ecc/android/sdk/demoapp/ui/screen/transactions/view/TransactionsView.kt:46]: this should be done in VM, never in UI too, but just copied the logic

File: demo-app/src/main/java/cash/z/ecc/android/sdk/demoapp/ui/screen/transactions/view/TransactionsView.kt [hunk 2]
```diff
@@ -23,63 +21,91 @@ import androidx.compose.runtime.Composable
...
+
+    val transactions = if (walletAddresses == null) {
+        emptyList()
+    } else {
+        synchronizer
+            .transactions
+            .collectAsStateWithLifecycle(initialValue = emptyList())
+            .value
+            .map { transactionOverview ->
+                transactionOverview.toTransactionState(
```
Comment (Milan-Nerevar) [inline demo-app/src/main/java/cash/z/ecc/android/sdk/demoapp/ui/screen/transactions/view/TransactionsView.kt:58]: mapping should never be done in UI but in VM instead

**Review by HonzaR — COMMENTED** [review]
Thank you for these changes. They bring a logic that enables us to extend it later easily.

I agree with all three of your comments inline. It's good to point this out. The Demo app itself is in the shadow of the Zashi wallet from the separate repository. That is the showcase project that the other wallet creates should follow. The Demo app is a testing tool for the SDK APIs. That's why the codebase itself does not necessarily need to follow all the best-practices. We can file this as a potential follow-up and improve it later.

Would you like to sync it with the original SDK repo state (we pushed some commits, I think)? And possibly resolve the CI actions failures.

**Review by HonzaR — APPROVED** [review]
This is a good improvement in our Demo app transaction list, which we can build upon quickly later. Thank you.

**Pull request merged**